File: Leetcode-942.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> ret;
        int i=0;
        int j=S.length();
        for(int s=0;s<S.length();s++)
        {
            if(S[s]=='I')
            {
                ret.push_back(i);
                i++;
            }
            else if(S[s]=='D')
            {
                ret.push_back(j);
                j--;
            }
        }
        ret.push_back(i);
        return ret;
    }
};

File: Leetcode-942.py
class Solution(object):
    def diStringMatch(self, S):
        """
        :type S: str
        :rtype: List[int]
        """
        i=0
        j=len(S)
        ret=[]
        for s in S:
            if(s=='I'):
                ret.append(i)
                i+=1
            if(s=='D'):
                ret.append(j)
                j-=1
        ret.append(i)
        return ret
